### PR TITLE
Mourning Chorus resolution pruning

### PR DESCRIPTION
**Pull request by Addisonstumpf**
Prune Mourning Chorus resolution options to only display defended locations (for rallying), and to only display the elites corresponding to infected locations (for elimination).

### CODE REVIEW

**Comment by zzorba** [comment]
I'm not totally sure, but I suspect we need to version this change.

I think despite the fact I have IDs, I'm storing choices based on indexes (bad, but hard to change at this point). So removing indexes conditionally might break this for existing campaigns.

If youve found otherwise I'd believe it, but the nagging voice in the back of my head says versioning this is safer.

**Comment by Addisonstumpf** [comment]
"safe than sorry" seems like a good rule for that kind of thing, sure. Nowhere near "having a personal understanding of versioning", let alone "opposing your understanding of versioning", ha.

**Comment by Addisonstumpf** [comment]
similar stuff needs to happen in (at least) Afternoon War, should probably lump those together instead of repeatedly running up the versioning. Will undraft when ready.

File: campaigns/zcp/mourning_chorus.json [hunk 1]
```diff
@@ -2005,10 +2005,277 @@
...
+          {
+            "boolCondition": true,
+            "steps": [
+              "record_rallied"
+            ]
+          },
+          {
+            "boolCondition": false,
+            "steps": [
+              "record_rallied_legacy"
```
Comment (zzorba) [inline campaigns/zcp/mourning_chorus.json:2026]: So the old one should keep the name "record_rallied" while the new one will need the new name ("record_rallied_new") -- basically we want to maintain the ID for the old version on the input.

File: campaigns/zcp/mourning_chorus.json [hunk 2]
```diff
@@ -2168,10 +2435,233 @@
...
+      "type": "branch",
+      "condition": {
+        "type": "campaign_data",
+        "campaign_data": "version",
+        "min_version": 4,
+        "options": [
+          {
+            "boolCondition": true,
+            "steps": [
+              "record_eliminated_elites"
```
Comment (zzorba) [inline campaigns/zcp/mourning_chorus.json:2450]: Same here, old version should keep the original name, new one needs a new name.

File: campaigns/zcp/the_afternoon_war.json [hunk 3]
```diff
@@ -1158,8 +1158,299 @@
...
+      "hidden": true,
+      "condition": {
+        "type": "campaign_data",
+        "campaign_data": "version",
+        "min_version": 4,
+        "options":[
+          {
+            "boolCondition": false,
+            "steps": [
+              "record_eliminated_elites_legacy"
```
Comment (zzorba) [inline campaigns/zcp/the_afternoon_war.json:1173]: We should flip these names, same as above.

**Comment by Addisonstumpf** [comment]
Checked, I had changed old names in the previous round of Mourning Chorus changes. Updated to A, revert that, and B, rename the new parts to use suffixes instead of prefixes. This feels like it probably also sets off a versioning problem, open to suggestions for how to minimize the wound here.

**Pull request merged**